### PR TITLE
added dropdown to select between http and https

### LINKED ISSUE #802: Getting error while adding my server url in sign in your server page.
- Your Rocket.Chat Experimental app version: 1.11.0
- Your Rocket.Chat server version: 0.74.3
- Device (or Simulator) you're running with: Device

**The app isn't connecting to your server?**
Make sure your server supports WebSocket. These are the minimum requirements for Apache 2.4 and Nginx 1.3 or greater.

Getting error Oops! The URL you entered is invalid. Check it and try again, please!

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
@RocketChat/ReactNative

Closes  #802

WhatsApp Image 2019-04-13 at 1 08 06 PM

### CODE REVIEW

File: app/views/NewServerView.js [hunk 1]
```diff
@@ -128,21 +134,14 @@ export default class NewServerView extends LoggedView {
...
-
-		if (/^(https?:\/\/)?(((\w|[0-9])+(\.(\w|[0-9-_])+)+)|localhost)(:\d+)?$/.test(url)) {
-			if (/^localhost(:\d+)?/.test(url)) {
-				url = `http://${ url }`;
-			} else if (/^https?:\/\//.test(url) === false) {
-				url = `https://${ url }`;
-			}
-		}
-
+		url = protocolType + url;
```
Comment (IlarionHalushka) [inline app/views/NewServerView.js]: prefer using templates instead of concatenation

File: app/views/NewServerView.js [hunk 2]
```diff
@@ -182,17 +181,30 @@ export default class NewServerView extends LoggedView {
...
-							containerStyle={styles.inputContainer}
-							placeholder={defaultServer}
-							value={text}
-							returnKeyType='send'
-							onChangeText={this.onChangeText}
-							testID='new-server-view-input'
-							onSubmitEditing={this.submit}
-							clearButtonMode='while-editing'
-						/>
+						<View
```
Comment (IlarionHalushka) [inline app/views/NewServerView.js]: might consider moving this View to a separate function for a better readability 🤔

File: app/views/NewServerView.js [hunk 3]
```diff
@@ -182,17 +181,30 @@ export default class NewServerView extends LoggedView {
...
+							style={styles.inputContainer}
+						>
+							<Picker
+								mode='dropdown'
+								selectedValue={protocolType}
+								onValueChange={this.handlePickerValueChange}
+								style={styles.picker}
+							>
+								<Picker.Item label='https://' value='https://' />
```
Comment (IlarionHalushka) [inline app/views/NewServerView.js]: for repeating strings it's better to define constants `protocol.HTTP` and `protocol.HTTPS`

File: app/views/NewServerView.js [hunk 4]
```diff
@@ -35,27 +35,25 @@ const styles = StyleSheet.create({
 		alignSelf: 'center'
 	},
 	inputContainer: {
+		flex: 1,
 		marginTop: 25,
-		marginBottom: 15
-	},
-	input: {
```
Comment (IlarionHalushka) [inline app/views/NewServerView.js]: not sure why you changed the styles and removed sharedStyles?

File: app/views/NewServerView.js (hunk 1 above)
Comment (pranavpandey1998official) [reply app/views/NewServerView.js]: cool

Comment (pranavpandey1998official) [reply app/views/NewServerView.js]: maybe += is better

Comment (IlarionHalushka) [reply app/views/NewServerView.js]: not critical at all, I just prefer using templates when working with strings (and templates are used all around in this project)

File: app/views/NewServerView.js (hunk 4 above)
Comment (pranavpandey1998official) [reply app/views/NewServerView.js]: input style was not being used previously, I just have created a new style here and named it input

**Comment by diegolmello** [comment]
I don't think this is a good user experience.
And we shouldn't help users to use insecure servers.

**Comment by pranavpandey1998official** [comment]
@diegolmello then I think we must inform them to use secure server

**Comment by Tom-Development** [comment]
I believe this is still an issue, on the react mobile app we cannot still join servers without HTTPS. 

This is not any good forcing people to use HTTPS sites when in our case are on local testing environments. There should still be an option to change between the two.

**Pull request closed without merging**